common/remote/sockfile.cpp

```diff
@@ -1256,6 +1317,8 @@ protected: friend class CRemoteFileIO;
         : filename(_filename)
     {
         ep = _ep;
+        //Determine whether or not client should use SSL
+        querySecuritySettings(&useSSL, &configepPort, &clientCertificate.certificate, &clientCertificate.privateKey);
```

Comment:
Agreed, it is unlikely which means it definitely will happen. Moved the call to querysecuritySettings to new static method, ensuring only done once and always available.